src/api/fetch.js

```diff
@@ -0,0 +1,13 @@
+function handleSuccess(res){
+    if(!res.ok){
+        throw new Error(`status code ${res.status}  ${res.statusText}`)
+    }
+    return res.json();
+}
+
+function fetchData(url){
```

Comment:
keep a separate function for different api calls, keep the endpoints here itself except IDs which you can accept it as an argument